barretenberg/cpp/src/barretenberg/api/api_client_ivc.cpp

```diff
@@ -210,6 +212,46 @@ void ClientIVCAPI::write_solidity_verifier([[maybe_unused]] const Flags& flags,
     throw_or_abort("API function contract not implemented");
 }
 
+bool ClientIVCAPI::check_ivc(const std::filesystem::path& input_path)
+
+{
+    init_bn254_crs(1 << CONST_PG_LOG_N);
+    init_grumpkin_crs(1 << CONST_ECCVM_LOG_N);
+    PrivateExecutionSteps steps;
+    steps.parse(PrivateExecutionStepRaw::load_and_decompress(input_path));
+
+    for (auto [program, precomputed_vk, function_name] :
+         zip_view(steps.folding_stack, steps.precomputed_vks, steps.function_names)) {
+        if (precomputed_vk == nullptr) {
+            info("FAIL: Expected precomputed vk for function ", function_name);
+            return false;
+        }
+        std::shared_ptr<ClientIVC::DeciderProvingKey> proving_key = get_acir_program_decider_proving_key(program);
+        auto computed_vk = std::make_shared<ClientIVC::MegaVerificationKey>(proving_key->proving_key);
```

Comment:
Hmm it looks like this is missing a couple of the components of a Mega VK. Its nice to have the nice prints with labels etc but it would also be more robust to use the `==` operator. Maybe we can update that operator to print nicely or something?

Reply:
In the mean time this is much better than nothing